src/components/Settings/MaxSlippageSettings/index.tsx

```diff
@@ -0,0 +1,159 @@
+import { Trans } from '@lingui/macro'
+import { Percent } from '@uniswap/sdk-core'
+import Expand from 'components/Expand'
+import QuestionHelper from 'components/QuestionHelper'
+import Row, { RowBetween } from 'components/Row'
+import React, { useState } from 'react'
+import { useUserSlippageTolerance } from 'state/user/hooks'
+import styled from 'styled-components/macro'
+import { ThemedText } from 'theme'
+
+import { Input, InputContainer } from '../Input'
+
+enum SlippageError {
+  InvalidInput = 'InvalidInput',
+}
+
+const Option = styled(Row)<{ isActive: boolean }>`
+  width: auto;
+  cursor: pointer;
+  padding: 6px 12px;
+  text-align: center;
+  gap: 4px;
+  border-radius: 12px;
+  background: ${({ isActive, theme }) => (isActive ? theme.backgroundInteractive : 'transparent')};
+  pointer-events: ${({ isActive }) => isActive && 'none'};
+`
+
+const Switch = styled(Row)`
+  width: auto;
+  padding: 4px;
+  border: 1px solid ${({ theme }) => theme.backgroundOutline};
+  border-radius: 16px;
+`
+
+const NUMBER_WITH_MAX_TWO_DECIMAL_PLACES = /^(?:\d*\.\d{0,2}|\d+)$/
+
+export default function MaxSlippageSettings({ placeholder }: { placeholder: Percent }) {
+  const [userSlippageTolerance, setUserSlippageTolerance] = useUserSlippageTolerance()
+
+  // If user has previously entered a custom deadline, we want to show that value in the input field
+  // instead of a placeholder by defualt
+  const [slippageInput, setSlippageInput] = useState(
+    userSlippageTolerance !== 'auto' && !userSlippageTolerance.equalTo(placeholder)
```

Comment:
can the 'auto' be made into an enum instead

Reply:
Good idea, converted.